server/app/services/statuses/StatusService.java

```diff
@@ -0,0 +1,258 @@
+   *     available statuses for application review.
+   */
+  public ErrorAnd<StatusDefinitions, CiviFormError> appendStatus(
+      long programId, StatusDefinitions.Status status)
+      throws ProgramNotFoundException, DuplicateStatusException {
+    ProgramDefinition program = programService.getFullProgramDefinition(programId);
+    StatusDefinitions currentStatus =
+        appStatusesRepo.lookupActiveStatusDefinitions(program.adminName());
+    if (currentStatus.getStatuses().stream()
+        .filter(s -> s.statusText().equals(status.statusText()))
+        .findAny()
+        .isPresent()) {
+      throw new DuplicateStatusException(status.statusText());
+    }
+
+    ImmutableList<StatusDefinitions.Status> updatedStatuses =
+        status.defaultStatus().orElse(false)
+            ? unsetDefaultStatus(currentStatus.getStatuses(), Optional.empty())
+            : currentStatus.getStatuses();
+
+    appStatusesRepo.createOrUpdateStatusDefinitions(
+        program.adminName(),
+        new StatusDefinitions()
+            .setStatuses(
+                ImmutableList.<StatusDefinitions.Status>builder()
+                    .addAll(updatedStatuses)
+                    .add(status)
+                    .build()));
+    return ErrorAnd.of(appStatusesRepo.lookupActiveStatusDefinitions(program.adminName()));
+  }
+
+  private ImmutableList<StatusDefinitions.Status> unsetDefaultStatus(
+      List<StatusDefinitions.Status> statuses, Optional<String> exceptStatusName) {
+    return statuses.stream()
+        .<StatusDefinitions.Status>map(
+            status ->
+                exceptStatusName.map(name -> status.matches(name)).orElse(false)
+                    ? status
+                    : status.toBuilder().setDefaultStatus(Optional.of(false)).build())
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  /**
+   * Updates an existing status this is available for application reviews.
+   *
+   * @param programId The program to update.
+   * @param toReplaceStatusName The name of the status that should be updated.
+   * @param statusReplacer A single argument function that maps the existing status to the value it
+   *     should be updated to. The existing status is provided in case the caller might want to
+   *     preserve values from the previous status (e.g. localized text).
+   * @throws ProgramNotFoundException If the specified Program could not be found.
+   * @throws DuplicateStatusException If the updated status already exists in the list of available
+   *     statuses for application review.
+   */
+  public ErrorAnd<StatusDefinitions, CiviFormError> editStatus(
+      long programId,
+      String toReplaceStatusName,
+      Function<StatusDefinitions.Status, StatusDefinitions.Status> statusReplacer)
+      throws ProgramNotFoundException, DuplicateStatusException {
+    ProgramDefinition program = programService.getFullProgramDefinition(programId);
+    StatusDefinitions currentStatus =
```

Comment:
Similarly here, I think all the callsites have a `ProgramDefinition` available so we don't need to fetch the program again inside the method, we can just pass in the admin name.

Reply:
done!